codegen/templates/chart/operator-deployment.yamltmpl

```diff
@@ -50,8 +49,13 @@ metadata:
     [[ $key ]]: [[ $value ]]
     [[- end ]]
   name: [[ $operator.Name ]]
-  [[- if $customNamespace ]]
-  namespace: [[ $customNamespace ]]
+  [[- if $operator.Values ]]
+{{- $[[ $operatorVar ]] := [[ (opVar $operator) ]]}}
+{{- if $[[ $operatorVar ]].namespace }}
+  namespace: {{ $[[ $operatorVar ]].namespace }}
+{{- else }}
+  namespace: {{ .Release.Namespace }}
+{{- end }}
```

Comment:
Nit: can this all be simplified using the `default` function? Not necessary but it makes the template a bit more concise. E.g. something like this

```
  namespace: {{ default .Release.Namespace [[ (opVar $operator) ]].namespace }}
```

Reply:
Thanks for the comment, this is much cleaner and works well. Just tested it on a local install.